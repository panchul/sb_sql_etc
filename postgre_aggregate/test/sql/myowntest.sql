-- MY OWN TEST, small table with ints
CREATE TABLE myownints1 (val int);

INSERT INTO myownints1(val)
SELECT *
FROM generate_series(0, 4);

SELECT * FROM myownints1;

SELECT median(val) FROM myownints1;

-- MY OWN TEST, medium table with ints
CREATE TABLE myownints_medium (val int);

INSERT INTO myownints_medium(val)
SELECT *
FROM generate_series(0, 10);

SELECT * FROM myownints_medium;

SELECT median(val) FROM myownints_medium;

-- MY OWN TEST, large table with ints
CREATE TABLE myownints2 (val int);

INSERT INTO myownints2(val)
SELECT *
FROM generate_series(0, 100000);

/* it's too large
SELECT * FROM myownints2; 
*/

SELECT median(val) FROM myownints2;

-- Now, working part of the other test
CREATE TABLE intvals(val int, color text);

-- Test empty table
SELECT median(val) FROM intvals;

-- Integers with odd number of values
INSERT INTO intvals VALUES
       (1, 'a'),
       (2, 'c'),
       (9, 'b'),
       (7, 'c'),
       (2, 'd'),
       (-3, 'd'),
       (2, 'e');

SELECT * FROM intvals ORDER BY val;
SELECT median(val) FROM intvals;

/*
-- Integers with NULLs and even number of values
INSERT INTO intvals VALUES
       (99, 'a'),
       (NULL, 'a'),
       (NULL, 'e'),
       (NULL, 'b'),
       (7, 'c'),
       (0, 'd');

SELECT * FROM intvals ORDER BY val;
SELECT median(val) FROM intvals;
*/

-- Text values
CREATE TABLE textvals(val text, color int);

INSERT INTO textvals VALUES
       ('erik', 1),
       ('mat', 3),
       ('rob', 8),
       ('david', 9),
       ('lee', 2);

SELECT * FROM textvals ORDER BY val;
SELECT median(val) FROM textvals;



-- Test large table with timestamps
CREATE TABLE timestampvals (val timestamptz);

INSERT INTO timestampvals(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);

SELECT median(val) FROM timestampvals;

