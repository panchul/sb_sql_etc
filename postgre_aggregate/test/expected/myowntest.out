-- MY OWN TEST, small table with ints
CREATE TABLE myownints1 (val int);
INSERT INTO myownints1(val)
SELECT *
FROM generate_series(0, 4);
SELECT * FROM myownints1;
 val 
-----
   0
   1
   2
   3
   4
(5 rows)

SELECT median(val) FROM myownints1;
 median 
--------
      2
(1 row)

-- MY OWN TEST, medium table with ints
CREATE TABLE myownints_medium (val int);
INSERT INTO myownints_medium(val)
SELECT *
FROM generate_series(0, 10);
SELECT * FROM myownints_medium;
 val 
-----
   0
   1
   2
   3
   4
   5
   6
   7
   8
   9
  10
(11 rows)

SELECT median(val) FROM myownints_medium;
 median 
--------
      5
(1 row)

-- MY OWN TEST, large table with ints
CREATE TABLE myownints2 (val int);
INSERT INTO myownints2(val)
SELECT *
FROM generate_series(0, 100000);
/* it's too large
SELECT * FROM myownints2; 
*/
SELECT median(val) FROM myownints2;
 median 
--------
  50000
(1 row)

-- Now, working part of the other test
CREATE TABLE intvals(val int, color text);
-- Test empty table
SELECT median(val) FROM intvals;
 median 
--------
       
(1 row)

-- Integers with odd number of values
INSERT INTO intvals VALUES
       (1, 'a'),
       (2, 'c'),
       (9, 'b'),
       (7, 'c'),
       (2, 'd'),
       (-3, 'd'),
       (2, 'e');
SELECT * FROM intvals ORDER BY val;
 val | color 
-----+-------
  -3 | d
   1 | a
   2 | c
   2 | e
   2 | d
   7 | c
   9 | b
(7 rows)

SELECT median(val) FROM intvals;
 median 
--------
      2
(1 row)

/*
-- Integers with NULLs and even number of values
INSERT INTO intvals VALUES
       (99, 'a'),
       (NULL, 'a'),
       (NULL, 'e'),
       (NULL, 'b'),
       (7, 'c'),
       (0, 'd');

SELECT * FROM intvals ORDER BY val;
SELECT median(val) FROM intvals;
*/
-- Text values
CREATE TABLE textvals(val text, color int);
INSERT INTO textvals VALUES
       ('erik', 1),
       ('mat', 3),
       ('rob', 8),
       ('david', 9),
       ('lee', 2);
SELECT * FROM textvals ORDER BY val;
  val  | color 
-------+-------
 david |     9
 erik  |     1
 lee   |     2
 mat   |     3
 rob   |     8
(5 rows)

SELECT median(val) FROM textvals;
 median 
--------
 lee
(1 row)

-- Test large table with timestamps
CREATE TABLE timestampvals (val timestamptz);
INSERT INTO timestampvals(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);
SELECT median(val) FROM timestampvals;
            median            
------------------------------
 Thu Jan 01 13:53:20 1970 PST
(1 row)

